### ISSUE TITLE
GetById queries are slow

### ISSUE THREAD

**Issue by senseicz** [Crokus/documentdb-repo#2]
Hi, 
I'm having strange issue with performance of GetById queries - while any other query takes no longer than 1 sec (this is locally deployed app connecting to Azure), GetById query takes 15+ seconds.
So far I'm working with relatively small dataset (between 1-2 thousands documents). I am using GUIDs for Id property.

I tried to play a bit with indexing but DocumentDB will not allow me to directly index Id field so I'm using:

```
{
      "path": "/*",
      "indexes": [
        {
          "kind": "Hash",
          "dataType": "String",
          "precision": -1
        },
        {
          "kind": "Hash",
          "dataType": "Number",
          "precision": 7
        }
      ]
    }
```

In my code, I declare my repo like this:

`_repoAggregationLog = new DocumentDbRepository<AggregationLog>(client, databaseId, () => "vd", p => p.Id);`

and it is declared as singleton for application lifetime, i.e. this is executed only once, not with every query.

Any idea what I can be doing wrong? Or is it normal that queries by document ID takes so long?

Many thanks, 

Antonin

**Comment by senseicz** [comment]
After some additional investigation, this runs way faster than original GetByIdAsync():

```
public async Task<T> GetByIdAsync(string id)
{
     var retVal = await GetDocumentByIdAsync(id);
     return (T)(dynamic)retVal;
}
```

**Comment by Crokus** [comment]
Hi,

Sorry for late response.

Did you have any benchmark results? Unfortunately, right now I'm too busy to check it by myself but I'll try during the weekend. Thanks!

**Comment by senseicz** [comment]
Hi, 
yes, I just did a quick unit test. Right now, I'm using S1 collection with 1602 documents. As I mentioned above, I use GUIDs as ID property. Warm-up queries are not measured and I am reusing same instance of DocumentClient for queries. Method 1 is the one from my previous post, Method 2 is original one. I ran 5 cycles:

Method 1 took (ms): 110
Method 2 took (ms): 26885
Method 1 took (ms): 81
Method 2 took (ms): 22222
Method 1 took (ms): 89
Method 2 took (ms): 23304
Method 1 took (ms): 80
Method 2 took (ms): 20822
Method 1 took (ms): 82
Method 2 took (ms): 22885

I've been using this new version of GetByIdAsync() for few days and so far all seems to be working fine, however my code is not yet at production level - so far only local testing.
